pkg/convert/xliff12.go

```diff
@@ -27,27 +30,43 @@ type bodyElement struct {
 }
 
 type transUnit struct {
-	ID     string `xml:"id,attr"`
-	Source string `xml:"source"`
-	Note   string `xml:"note,omitempty"`
+	ID     string `xml:"id,attr"`          // messages.messages[n].ID
+	Source string `xml:"source"`           // messages.messages[n].Message (if no target language is set)
+	Target string `xml:"target,omitempty"` // messages.messages[n].Message (if target language is set)
+	Note   string `xml:"note,omitempty"`   // messages.messages[n].Description
+	// No unified standard about storing fuzzy values
 }
 
 // FromXliff12 converts serialized data from the XML data in the XLIFF 1.2 format into a model.Messages struct.
 func FromXliff12(data []byte) (model.Messages, error) {
 	var xlf xliff12
 	if err := xml.Unmarshal(data, &xlf); err != nil {
-		return model.Messages{}, fmt.Errorf("unmarshal XLIFF 1.2 formatted XML into xliff12 struct: %w", err)
+		return model.Messages{}, fmt.Errorf("unmarshal xliff12: %w", err)
 	}
 
 	messages := model.Messages{
 		Language: xlf.File.SourceLanguage,
 		Messages: make([]model.Message, 0, len(xlf.File.Body.TransUnits)),
 	}
 
+	// Check if file has a target language set
+	isTranslated := xlf.File.TargetLanguage != language.Und
+	if isTranslated {
+		messages.Language = xlf.File.TargetLanguage
+	}
+
+	getMessage := func(t transUnit) string {
+		if isTranslated {
+			return t.Target
+		}
+
+		return t.Source
+	}
```

Comment:
You can improve efficiency a lot. Right now, for each message it does `if` statement. The function is a first-class citizen in Golang - use it.

```suggestion
	getMessage := func(t transUnit) string { return t.Source}
	
	if isTranslated {
		getMessage = func(t transUnit) string { return t.Target }
	}
```